src/stream/src/executor/sink.rs

```diff
@@ -184,6 +184,7 @@ impl<F: LogStoreFactory> SinkExecutor<F> {
         let re_construct_with_sink_pk = need_advance_delete
             && self.sink_param.sink_type == SinkType::Upsert
             && !self.sink_param.downstream_pk.is_empty();
+        let is_blackhole = self.sink_param.connector() == Some("blackwhole");
```

Comment:
```suggestion
        let is_blackhole = self.sink_param.connector() == Some("blackhole");
```
?

Reply:
Can we use match the variant of `self.sink` instead (similar to `is_sink_into_table()`)? The current way looks fragile.